### ISSUE TITLE
Debugger occasionally fails to step into a x86 Linux binary

### ISSUE THREAD

**Issue by xusheng6** [Vector35/debugger#386]
When we single-step consecutively a x86 Linux binary, we sometimes get an error like this:
```
>>> si
Resume request failed - process still running.
```

Note this does NOT require stepping very rapidly -- stepping in a normal speed can also lead to the problem.

Weird enough, this issue happens after https://github.com/Vector35/debugger/commit/c7bc871ca72dae037d20191dc4ed452b0eb27abf, which, on the surface, has nothing to do with the issue

**Comment by xusheng6** [comment]
I tried to use the test script at https://github.com/xusheng6/lldb_test to reproduce the issue, but it works correctly. Also this only affects x86 binaries, not x64 ones.

The related code in LLDB is:

```C
Status Process::Resume() {
  Log *log(GetLog(LLDBLog::State | LLDBLog::Process));
  LLDB_LOGF(log, "Process::Resume -- locking run lock");
  if (!m_public_run_lock.TrySetRunning()) {
    Status error("Resume request failed - process still running.");
    LLDB_LOGF(log, "Process::Resume: -- TrySetRunning failed, not resuming.");
    return error;
  }
....
```

It is likely that some part of the debugger causes the lock `m_public_run_lock` to be taken, thus affecting the resume operation

**Comment by xusheng6** [comment]
This is caused by the same problem as https://github.com/Vector35/debugger/issues/384

**Comment by xusheng6** [comment]
Fixed in 6d4820960db1db220eac788f18e535d9617370eb
